File: fib.pl
# perl fib.pl - 345.10891 seconds
use warnings;
use strict;
use Time::HiRes qw( time );

sub fib {
   my ($n) = @_;
   if($n <= 0) {
       return 0;
   }
   if($n <= 2) {
       return 1;
   }
   return fib($n-1) + fib($n-2);
}

$| = 1; # autoflush on
my $start = time();
foreach (1..42) {
    my $result = fib($_);
    print("$result ")
}
my $end = time();
printf("\n%f s\n", ($end - $start));

File: fib.c
// gcc -o fibc fib.c
// Time taken 11 seconds 654 milliseconds
#include <stdio.h>
#include <time.h>

int fib(int n) {
    if(n <= 2) {
        return 1;
    }
    return fib(n-2) + fib(n-1);
}

int main() {
    clock_t start = clock(), diff;

    for(int i = 1; i <= 42; i++) {
        printf("%d ", fib(i));
        fflush(stdout);
    }
    diff = clock() - start;
    int msec = diff * 1000 / CLOCKS_PER_SEC;
    printf("\nTime taken %d seconds %d milliseconds\n", msec/1000, msec%1000);
    return 0;
}
